mvrx/src/main/kotlin/com/airbnb/mvrx/RealMvRxStateStore.kt

```diff
@@ -126,15 +119,12 @@ class RealMvRxStateStore<S : Any>(initialState: S) : MvRxStateStore<S> {
         flushQueues()
     }
 
-    /**
-     * Coalesce all updates on the setState queue and clear the queue.
-     */
     private fun flushSetStateQueue() {
-        val blocks = jobs.dequeueAllSetStateBlocks() ?: return
-
-        blocks
-            .fold(state) { state, reducer -> state.reducer() }
-            .run { if (this != state) subject.onNext(this) }
+        val block = jobs.dequeueSetStateBlock() ?: return
+        val newState = block(state)
+        if (newState != state) {
+            subject.onNext(newState)
+        }
```

Comment:
@gpeal @BenSchwab @elihart I think we still need to flush the whole `setStateQueue` here to prioritize reducers over accessors, to avoid a race condition we discussed a while back. 

To avoid coalescing the change, I think we can simply emit the new value once for every reducers in the queue.

```
private fun flushSetStateQueue() {
	        val blocks = jobs.dequeueAllSetStateBlocks() ?: return
                for (block in blocks) {
                    val newState = state.reducer()
                    if (newState != state) { 
                          subject.onNext(newState) 
                    }
                }
```

Reply:
Let me put up a PR today and also add some comment on this race condition.